go/tasks/plugins/array/k8s/task.go

```diff
@@ -0,0 +1,232 @@
+package k8s
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	idlCore "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array/arraystatus"
+	arrayCore "github.com/lyft/flyteplugins/go/tasks/plugins/array/core"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array/errorcollector"
+	"github.com/lyft/flytestdlib/bitarray"
+	errors2 "github.com/lyft/flytestdlib/errors"
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/storage"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sTypes "k8s.io/apimachinery/pkg/types"
+)
+
+type Task struct {
+	LogLinks         []*idlCore.TaskLog
+	State            *arrayCore.State
+	NewArrayStatus   *arraystatus.ArrayStatus
+	Config           *Config
+	ChildIdx         int
+	MessageCollector *errorcollector.ErrorMessageCollector
+}
+
+type TaskStatus int8
+
+const (
+	Success TaskStatus = iota
+	Error
+	Waiting
+	ReturnState
```

Comment:
it's not immediately clear what this TaskStatus represents. can you add a comment explaining 'ReturnState'?

Reply:
So there are essentially two state machines one for the entire Job and one for the subtasks. There are cases where we would like to return the entire job state based off of the outcome of a subtask. I couldn't come up with a great name. Any ideas?(will add a comment after deciding on name)

Reply:
I think it's also confusing that we return Success from Launch when we successfully launch and we return the same Success from Monitor() when the task finishes successfully.
Maybe you just need LaunchResult and MonitorResult (two separate enums)

Reply:
Updated to use two seperate enums. Still unsure about the name for `returnState`. Any ideas for a better name?